### PR TITLE
feat: expose raw devtools protocol connection

### LINKED ISSUE #31: Expose raw protocol in the API
While using puppeteer there are still occasions where I want to use the raw protocol.
I'd certainly want to do this against the Page, and potentially the browser as well.

I want to send methods and get responses. Additionally, I want to listen for specific events.

Not sure of the right API and if it needs to be transactional as to not mixup state.

### PR DESCRIPTION
**Pull request by aslushnikov**
This patch introduces `target.createCDPSession` method that
allows directly communicating with the target over the
Chrome DevTools Protocol.

Fixes #31.

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -2166,6 +2174,11 @@ Contains the URL of the response.
 
 ### class: Target
 
+#### target.createCDPSession()
+- returns: <[CDPSession]>
```
Comment (JoelEinbinder) [inline docs/api.md]: Promise to session? And the link doesn't work.

File: docs/api.md [hunk 2]
```diff
@@ -2166,6 +2174,11 @@ Contains the URL of the response.
...
+
+Creates a *C*hrome *D*evtools *P*rotocol session attached to the target.
```
Comment (JoelEinbinder) [inline docs/api.md]: Let's not have the letters in italics.

File: docs/api.md [hunk 3]
```diff
@@ -2192,6 +2192,7 @@ Identifies what kind of target this is. Can be `"page"`, `"service_worker"`, or
 #### target.url()
 - returns: <[string]>
 
+
```
Comment (JoelEinbinder) [inline docs/api.md:2195]: This newline is in the wrong place.

Comment (aslushnikov) [reply docs/api.md:2195]: Added another one.

**Review by JoelEinbinder — APPROVED** [review]

**Pull request merged**